client/era/era_structs.go

```diff
@@ -0,0 +1,139 @@
+package era
+
+type Clusteravailability struct {
+	Nxclusterid  *string `json:"nxClusterId,omitempty"`
+	Datecreated  *string `json:"dateCreated,omitempty"`
+	Datemodified *string `json:"dateModified,omitempty"`
+	Ownerid      *string `json:"ownerId,omitempty"`
+	Status       *string `json:"status,omitempty"`
+	Profileid    *string `json:"profileId,omitempty"`
+}
+
+// ListProfile response
+type ListProfileResponse struct {
+	ID                  *string                `json:"id,omitempty"`
+	Name                *string                `json:"name,omitempty"`
+	Description         *string                `json:"description,omitempty"`
+	Status              *string                `json:"status,omitempty"`
+	Datecreated         *string                `json:"dateCreated,omitempty"`
+	Datemodified        *string                `json:"dateModified,omitempty"`
+	Owner               *string                `json:"owner,omitempty"`
+	Enginetype          *string                `json:"engineType,omitempty"`
+	Type                *string                `json:"type,omitempty"`
+	Topology            *string                `json:"topology,omitempty"`
+	Dbversion           *string                `json:"dbVersion,omitempty"`
+	Systemprofile       bool                   `json:"systemProfile,omitempty"`
+	Latestversion       *string                `json:"latestVersion,omitempty"`
+	Latestversionid     *string                `json:"latestVersionId,omitempty"`
+	Versions            []*Versions            `json:"versions,omitempty"`
+	Assocdbservers      []interface{}          `json:"assocDbServers,omitempty"`
+	Assocdatabases      []*string              `json:"assocDatabases,omitempty"`
+	Nxclusterid         *string                `json:"nxClusterId,omitempty"`
+	Clusteravailability []*Clusteravailability `json:"clusterAvailability,omitempty"`
+}
+
+type ProfileListResponse []ListProfileResponse
+
+type Propertiesmap struct {
+	DefaultContainer *string `json:"DEFAULT_CONTAINER"`
+	MaxVdiskSize     *string `json:"MAX_VDISK_SIZE"`
+}
+
+type Properties struct {
+	RefID       interface{} `json:"ref_id,omitempty"`
```

Comment:
is the return type of RefID not constant ? thats why we used interface here ?

Reply:
changed it to "string" type .